src/process/process.cc

```diff
@@ -77,45 +77,66 @@ namespace blind
             return sqrt(pow(a.x - b.x, 2) + pow(a.y - b.y, 2));
         }
 
-        cv::Point getDirection(const cv::Point &base, const int nb_labels,
-                               cv::Mat &centroids, cv::Mat &yellow_cones)
+        cv::Point getMiddle(cv::Point &red, cv::Point &yellow, int max_cols,
+                            int max_rows)
         {
+            cv::Point mid =
+                cv::Point((red.x + yellow.x) / 2, (red.y + yellow.y) / 2);
+            if (mid.x == 0 && mid.y == 0)
+                mid = cv::Point(max_cols / 2, max_rows / 2);
+            else if (red.x == 0 && red.y == 0)
+                mid = cv::Point(0, yellow.y);
+            else if (yellow.x == 0 && yellow.y == 0)
+                mid = cv::Point(max_cols, red.y);
+
+            return mid;
+        }
+        cv::Point getDirection(const int nb_labels, cv::Mat &centroids,
+                               cv::Mat &yellow_cones, cv::Mat &stats)
+        {
+            cv::Point base(yellow_cones.cols / 2, yellow_cones.rows);
             cv::Point red(0, 0);
+            int max_red_size = 0;
             cv::Point yellow(0, 0);
+            int max_yellow_size = 0;
             for (int i = 1; i < nb_labels; i++)
             {
-                cv::Point p(centroids.at<double>(i, 0),
-                            centroids.at<double>(i, 1));
-                auto val = yellow_cones.at<unsigned char>(p);
+                cv::Point curr_p(centroids.at<double>(i, 0),
+                                 centroids.at<double>(i, 1));
+                auto val = yellow_cones.at<unsigned char>(curr_p);
+                int size = stats.at<int>(i, cv::CC_STAT_AREA);
```

Comment:
```suggestion
                auto size = stats.at<int>(i, cv::CC_STAT_AREA);
```

Reply:
Why would auto be better than specifying the type for a simple one as 'int' ?

Reply:
Consistency